packages/amplication-server/Dockerfile

```diff
@@ -1,18 +1,25 @@
-FROM node:18.12.1-slim AS base
+FROM node:18.13.0-slim AS base
 
-# Required to run Prisma on ARM64 machines
-# https://github.com/prisma/prisma/issues/861#issuecomment-881992292
-RUN apt-get update
-RUN apt-get install -y openssl
+ARG component=server
+ARG user=${component}
+ARG uid=1001
+ARG gid=$uid
 
-WORKDIR /app
+# Required to run Prisma on ARM64 machine: https://github.com/prisma/prisma/issues/861#issuecomment-881992292
+RUN apt-get update && apt-get install -y openssl
 
-COPY ./dist/packages/amplication-server/package*.json ./
+WORKDIR /workspace/packages/${user}
 
-RUN npm i --omit=dev
+COPY ./dist/packages/amplication-${component}/package*.json .
+RUN npm install --omit-dev
 
-COPY ./dist/packages/amplication-server .
+COPY ./dist/packages/amplication-${component} .
 
+RUN groupadd --gid ${gid} ${user}
+RUN useradd --uid ${uid} --gid ${gid} -m ${user}
+RUN chown -R ${uid}:${gid} /workspace/packages/${user}
+
+USER ${user}
 
 ENV PORT=3333
 EXPOSE $PORT
```

Comment:
amplication-server container doesn't start due to permissions.

```sh
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] MailModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] BillingModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] TopicModule dependencies initialized +1ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] CommitModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] SubscriptionModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] PluginInstallationModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] GraphQLModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] ResourceModule dependencies initialized +1ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] GitProviderModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] AccountModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] BuildModule dependencies initialized +1ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] ProjectModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] WorkspaceModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM     LOG [InstanceLoader] AuthModule dependencies initialized +0ms
[Nest] 1  - 01/25/2023, 11:53:41 AM   ERROR [ServerKafka] ERROR [BrokerPool] Failed to connect to seed broker, trying another broker from the list: Failed to connect: Port should be >= 0 and < 65536. Received type number (NaN). {"timestamp":"2023-01-25T11:53:41.464Z","logger":"kafkajs","retryCount":0,"retryTime":303}
node:internal/process/promises:288
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[Error: EACCES: permission denied, open '../.server-id'] {
  errno: -13,
  code: 'EACCES',
  syscall: 'open',
  path: '../.server-id'
}

Node.js v18.13.0
```

To test it locally I suggest to run `npx nx docker:build amplication-server` and `docker run --env-file packages/amplication-server/.env amplication/server`

Reply:
this potentially affects other container images too but I have not check them yet

Reply:
the problem here was the change ownership command that was a directory to far down, it was changed to:
```
RUN chown -R ${uid}:${gid} /workspace/packages/
```